config/eslint.config.js

```diff
@@ -0,0 +1,173 @@
+module.exports = {
+  extends: ['airbnb', 'plugin:prettier/recommended'],
+  plugins: ['import', 'jsx-a11y', 'react', 'react-hooks'],
+  parserOptions: {
+    ecmaVersion: 'latest',
+  },
+  env: {
+    browser: true,
+    node: true,
+  },
+  rules: {
+    // eslint
+    'no-unused-expressions': [
+      'error',
+      {
+        allowShortCircuit: true,
+        allowTernary: true,
+        allowTaggedTemplates: true,
+      },
+    ],
+    'no-console': [
+      'error',
+      {
+        allow: ['warn', 'error', 'info'],
+      },
+    ],
+    'no-plusplus': [
+      'error',
+      {
+        allowForLoopAfterthoughts: true,
+      },
+    ],
+    'no-use-before-define': [
+      'error',
+      {
+        functions: false,
+        classes: false,
+        variables: false,
+      },
+    ],
+    'no-restricted-exports': [
+      'error',
+      {
+        restrictedNamedExports: ['then'],
+      },
+    ],
+    'default-case': [
+      'warn',
+      {
+        commentPattern: '^no default$',
+      },
+    ],
+    'arrow-body-style': 'off',
+    'prefer-arrow-callback': 'off',
+
+    // eslint-plugin-import https://github.com/import-js/eslint-plugin-import
+    'import/no-unresolved': 'off',
+    'import/prefer-default-export': 'off',
+
+    // eslint-plugin-jsx-a11y https://github.com/jsx-eslint/eslint-plugin-jsx-a11y
+    'jsx-a11y/label-has-associated-control': [
+      'error',
+      {
+        assert: 'either',
+      },
+    ],
+
+    // eslint-plugin-react https://github.com/jsx-eslint/eslint-plugin-react
+    'react/no-direct-mutation-state': 'error',
+    'react/no-multi-comp': [
+      'error',
+      {
+        ignoreStateless: false,
+      },
+    ],
+    'react/react-in-jsx-scope': 'off',
```

Comment:
d'apres la doc eslint, ils recommandent de faire comme ca : 
> If you are using the [new JSX transform from React 17](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports), you should disable this rule by extending [react/jsx-runtime](https://github.com/jsx-eslint/eslint-plugin-react/blob/8cf47a8ac2242ee00ea36eac4b6ae51956ba4411/index.js#L165-L179) in your eslint config (add "plugin:react/jsx-runtime" to "extends").

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md

Reply:
Idem besoin de le désactiver pour un autre projet mais ici pas utile.